sidecar/internal/service/reclaimspace.go

```diff
@@ -88,14 +116,37 @@ func (rs *ReclaimSpaceServer) NodeReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get pv: %v", err)
+	}
 
+	if pv.Spec.CSI == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "pv %q is not a CSI volume", pvName)
+	}
+	volID := pv.Spec.CSI.VolumeHandle
+	csiMode, err := accessmodes.ToCSIAccessMode(pv.Spec.AccessModes, true)
+	stPath := *rs.stagingPath + pvName + "/globalmount"
```

Comment:
This requires `rs.stagingPath` to end with a `/`, otherwise the path becomes invalid. Consider adding slashes with `fmt.Sprintf()` or use [`filepath.Join()`](https://pkg.go.dev/path/filepath#Join).

Reply:
ohh yeah.